File: apps/tv/modules/valence-glass/ios/ValenceGlassModule.swift
import ExpoModulesCore

/// Offers JavaScript a view that draws Liquid Glass behind what it holds.
public class ValenceGlassModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ValenceGlass")

    View(ValenceGlassView.self) {
      Prop("cornerRadius") { (view: ValenceGlassView, cornerRadius: Double) in
        view.cornerRadius = CGFloat(cornerRadius)
      }
    }
  }
}

File: apps/tv/modules/valence-glass/android/src/main/java/app/valence/tv/glass/ValenceGlassModule.kt
package app.valence.tv.glass

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/** Offers JavaScript a view that draws a pane of glass behind whatever it holds. */
class ValenceGlassModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ValenceGlass")

    View(ValenceGlassView::class) {
      Prop("cornerRadius") { view: ValenceGlassView, radius: Float -> view.cornerRadius = radius }
    }
  }
}
